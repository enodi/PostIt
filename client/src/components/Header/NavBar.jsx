import React from 'react';
import { Link, IndexLink } from 'react-router';

/**
 * NavBar Presentational Component
 *
 * @param {object} props
 *
 * @returns {jsx} an xml/html like syntax extension for
 * javascript
 */
const NavBar = () => (
  <nav>
    <div className="nav-wrapper #212121 grey darken-4">
      <IndexLink to="/" className="brand-logo">
        <span
          className="logo grey-text text-lighten-2"
        >PostIt
        </span>
      </IndexLink>
      <Link to="#"
        data-activates="mobile-demo"
        className="button-collapse">
        <i className="material-icons">menu</i>
      </Link>
      <ul className="right hide-on-med-and-down">
        <li><a href="http://postit-enodi.herokuapp.com/apidocs/#introduction">
        API Docs</a></li>
        <li><Link to="/account">My Account</Link></li>
      </ul>
      <ul className="side-nav" id="mobile-demo">
        <li><Link to="/account">My Account</Link></li>
      </ul>
    </div>
  </nav>
);

export default NavBar;
