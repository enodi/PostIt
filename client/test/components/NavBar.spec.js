import React from 'react';
import { shallow } from 'enzyme';

import NavBar from '../../src/components/Header/NavBar.jsx';

describe('NavBar Component', () => {
  it('should render NavBar Component', () => {
    const wrapper = shallow(<NavBar />);
    expect(wrapper).toMatchSnapshot();
  });
});

