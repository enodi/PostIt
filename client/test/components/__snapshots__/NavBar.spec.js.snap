// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Component should render NavBar Component 1`] = `
<nav>
  <div
    className="nav-wrapper #212121 grey darken-4"
  >
    <IndexLink
      className="brand-logo"
      to="/"
    >
      <span
        className="logo grey-text text-lighten-2"
      >
        PostIt
      </span>
    </IndexLink>
    <a
      className="button-collapse"
      data-activates="mobile-menu"
      href="#"
    >
      <i
        className="material-icons"
      >
        menu
      </i>
    </a>
    <ul
      className="right hide-on-med-and-down"
    >
      <li>
        <a
          href="http://postit-enodi.herokuapp.com/apidocs/#introduction"
        >
          API Docs
        </a>
      </li>
      <li>
        <Link
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/account"
        >
          My Account
        </Link>
      </li>
    </ul>
    <ul
      className="side-nav #000000 black hide-on-large-only"
      id="mobile-menu"
    >
      <li>
        <a
          className="white-text"
          href="http://postit-enodi.herokuapp.com/apidocs/#introduction"
        >
          API Docs
        </a>
      </li>
      <li>
        <Link
          className="white-text"
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/account"
        >
          My Account
        </Link>
      </li>
    </ul>
  </div>
</nav>
`;
